packages/hash/frontend/src/pages/[account-slug]/types/entity-type/delete-data-type-modal.tsx

```diff
@@ -0,0 +1,117 @@
+import { faClose } from "@fortawesome/free-solid-svg-icons";
+import {
+  Button,
+  FontAwesomeIcon,
+  IconButton,
+} from "@hashintel/hash-design-system";
+import { Box, Stack, Typography } from "@mui/material";
+import { bindDialog, PopupState } from "material-ui-popup-state/hooks";
+import { ReactNode } from "react";
+import { Modal } from "../../../../components/Modals/Modal";
+
+export interface DeleteDataTypeModalProps {
+  popupState: PopupState;
+  onClose?: () => void;
+  onDelete?: () => void;
+  dataTypeCount?: number;
+  arrayCount?: number;
+  propertyObjectCount?: number;
+}
+
+export const DeleteDataTypeModal = ({
+  popupState,
+  onClose,
+  onDelete,
+  dataTypeCount = 0,
+  arrayCount = 0,
+  propertyObjectCount = 0,
+}: DeleteDataTypeModalProps) => {
+  const countArray = [
+    { label: "data type", count: dataTypeCount },
+    { label: "array", count: arrayCount },
+    { label: "property object", count: propertyObjectCount },
+  ].filter(({ count }) => count);
+
+  return (
+    <Modal
+      {...bindDialog(popupState)}
+      contentStyle={(theme) => ({
+        p: "0px !important",
+        border: 1,
+        borderColor: theme.palette.gray[20],
+      })}
+    >
+      <>
+        <Box
+          sx={{
+            height: 56,
+            padding: 2.5,
+            borderBottom: ({ palette }) => `1px solid ${palette.gray[20]}`,
+          }}
+        >
+          <Typography
+            variant="largeTextLabels"
+            sx={{
+              fontSize: 16,
+              color: ({ palette }) => palette.gray[80],
+              fontWeight: 500,
+            }}
+          >
+            Remove array and its contents
+          </Typography>
+
+          <IconButton
+            sx={{ position: "absolute", top: 16, right: 16 }}
+            onClick={onClose}
+          >
+            <FontAwesomeIcon icon={faClose} />
+          </IconButton>
+        </Box>
+        <Box sx={{ padding: 3 }}>
+          <Box sx={{ marginBottom: 2, fontSize: 14, lineHeight: "18px" }}>
+            <Typography
+              variant="smallTextLabels"
+              sx={{ color: ({ palette }) => palette.gray[80] }}
+            >
+              This array contains
+              {countArray.reduce(
```

Comment:
As discussed on Tandem, something like this would be a lot easier to read

```ts
type CountItemProps = { label: string, count: number };

const CountItem = ({ label, count }: CountItemProps) => (
  <strong>{count} {label}{count > 1 ? <>s</> : null}</strong>
);

const CountGroup = ({ items }: { items: CountItemProps[] }) => {
  return items.map(
    (item, index) => (
      <Fragment key={item.label}>
        <CountItem {...item} />
        {items[index + 2] ? <>, </> : items[index + 1] ? <>and </> : <>.</>}
      </Fragment>
    )
  )
};

<CountGroup items={countArray} />
```